src/lib/ptree/ptree.lua

```diff
@@ -297,17 +301,85 @@ function Manager:compute_scheduling_for_worker(id, app_graph)
    return ret
 end
 
+local counters = {active={}, archived={}}
+
+local function is_counter (name)
+   return lib.basename(name):match("%.counter$")
+end
+local function read_counter (name)
+   if not counters.active[name] then counters.active[name] = counter.open(name) end
+   return counter.read(counters.active[name])
+end
+local function update_counter (name, val)
+   local c = assert(counters.active[name])
+   counter.add(c, val)
+end
+local function create_counter (name, val)
+   counters.active[name] = counter.create(name, val)
+end
+local function archive_counter (name)
+   local c = assert(counters.active[name])
+   local val = counter.read(c)
+   counter.delete(name)
+   counters.active[name] = nil
+   counters.archived[name] = counters.archived[name] and counter.set(name, val)
```

Comment:
many workers can accumulate their values into one archived counter.  e.g. you have 1234/foo/bar.counter and 2345/foo/bar.counter; the aggregate is the sum of all of these, plus the archived value for foo/bar (which can simply be a uint64_t[1] instead of a full counter; probably it doesn't need to go out to the filesystem).  When 1234 goes away you want to add its value to the archived counter.  WDYT?

Reply:
OK, I think I understand now better. It's also my understanding that when displaying the aggregated countries the actual value will be the sum of active aggregated counter and archived aggregated "counter".

Reply:
yeah i think that's the idea.